lib/tbot/config/bot.go

```diff
@@ -42,4 +48,145 @@ type Bot interface {
 
 	// Client retrieves the current auth client.
 	Client() auth.ClientI
+
+	// AuthenticatedUserClientFromIdentity returns a client backed by a specific
+	// identity.
+	AuthenticatedUserClientFromIdentity(ctx context.Context, id *identity.Identity, authServer string) (auth.ClientI, error)
```

Comment:
Does AuthenticatedUserClientFromIdentity interface function is really needed. Right no it only referring to bot.Log to internal Bot filed to create the auth Client ?

Reply:
Fair point, I'll move it into the `identity` package.

Reply:
Hrm, on second thought, we need to be able to mock this in tests, so promoting it to a top-level function or otherwise splitting it up while keeping the function mockable will require more of a redesign than I think is worth pursuing at the moment.